### DISCUSSION TITLE
GS1 Barcodes creation and encoding

### DISCUSSION THREAD

**Discussion by Kizuto3** [micjahn/ZXing.Net#418] [Q&A] [answered]
Hello

I have a few questions regarding GS1 Barcodes encoding:
1. How to create a GS1 QR Code? Currently I am using QRCodeWriter class and passing a dictionary with a GS1_FORMAT set to true to the encode() method as hints. But it appears that FNC1 symbol is only added to the beginning of a passed string. Is there a way to specify FNC1 symbol in the passed string as in GS1 DataMatrix and GS1 Code 128? I've looked through the source code but couldn't find anything that would help me. For example, I want to encode the next string: "{FNC1}11000123{FNC1}1010". What symbol do I have to use instead of "{FNC1}"?
2. Does DataMatrixWriter support GS1 DataMatrix? I've found that the symbol that I have to use to specify {FNC1} is 0x29. In the wiki article about GS1 DataMatrix the class that is used to create GS1 DataMatrix is a regular BarcodeWriter, so I just want to be sure that DataMatrixWriter is applicable in this case. Again, example would be the next string: "{FNC1}11000123{FNC1}1010". Is it correct to pass the string $"{char(29)}11000123{char(29)}1010" to the encode() method of DataMatrixWriter to get GS1 DataMatrix?

**Comment by micjahn** [comment, marked as answer]
1. If GS1_Format is set only the FNC1 character is placed at the first position. There is no other interpretation of "{FNC1}" placeholders implemented. You can try to use a similar string like it is used for the DataMatrix example here: https://github.com/micjahn/ZXing.Net/wiki/Generating-GS1-DataMatrix-and-GS1-Code128-barcodes
But I never tried to generate a GS1-QR myself before.
2. BarcodeWriter uses the DataMatrixWriter under the hood. You can use both of them like it is described in the wiki.

**Reply by ahmed-abdelrazek** [reply]
how can i add label to that?!
